src/esp.d.ts

```diff
@@ -0,0 +1,83 @@
+// notice_start
+/*
+ * Copyright 2015 Dev Shop Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+// notice_end
+
+declare module esp {
+
+    function observeEvent(eventName:String, observationStage?:string): any;
+
+    class Router {
+        addModel<T>(modelId:String, model:T, options) : void;

```

Comment:
It seems to only have the pre and post processor functions on it, which I
wanted to chat to you about first.  I was thinking we should create an
explicit object anyway for non-TS users.

On 15 March 2016 at 15:41, Keith Woods <notifications@github.com> wrote:

> In src/esp.d.ts
> <https://github.com/esp/esp-js/pull/82#discussion_r56185448>:
>
> > + *   http://www.apache.org/licenses/LICENSE-2.0
> > + *
> > + * Unless required by applicable law or agreed to in writing, software
> > + * distributed under the License is distributed on an "AS IS" BASIS,
> > + * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
> > + * See the License for the specific language governing permissions and
> > + * limitations under the License.
> > + */
> > +// notice_end
> > +
> > +declare module esp {
> > +
> > +    function observeEvent(eventName:String, observationStage?:string): any;
> > +
> > +    class Router {
> > +        addModel<T>(modelId:String, model:T, options) : void;
>
> options is optional, not sure if you need to specify optionality for TS
> at this point. Presumably options?:x.
>
> I'll have to have a look at the shape of 'options', I don't think there is
> an explicit object
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/esp/esp-js/pull/82/files#r56185448
>